### PR TITLE
Details section notification

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# Details section notification

Support for stackable notifications for position details section.

image
image
  
## Changes 👷‍♀️

- Extended `DetailsSection` component with `notifications` props,
- created `DetailsSectionNotification` component.
  
## How to test 🧪

This is just a component, it's not used anywhere. If you want to test it, find any place with `Details section` and try to add one of the props listed below:

```typescript
        notifications={[
          {
            closable: true,
            icon: 'wallet',
            title: 'You are viewing 0x373...48304 vault',
            type: 'notice',
          },
        ]}
```
```typescript
        notifications={[
          {
            closable: true,
            icon: 'liquidation',
            title: 'Your vault got liquidated',
            message: 'A liquidation has happened on this vault. Reclaim 2.20 ETH',
            link: {
              label: 'Reclaim collateral',
              action: () => alert('Action clicked'),
            },
            type: 'error',
          },
        ]}
```
```typescript
        notifications={[
          {
            icon: 'coins_cross',
            title: 'This notification can\'t be closes and doesn\'t have any action.',
            type: 'warning',
          },
          {
            closable: true,
            icon: 'liquidation',
            title: 'This notification has link instead of action.',
            link: {
              label: 'Go to homepage',
              url: '/',
            },
            type: 'notice',
          },
        ]}
```

### CODE REVIEW

File: components/DetailsSectionNotification.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      sx={{
+        position: 'relative',
+        p: 0,
+        borderTopLeftRadius: 'roundish',
+        borderTopRightRadius: 'roundish',
+        overflow: 'hidden',
+      }}
+    >
+      {notifications.map(({ closable, icon, link, message, title, type = 'notice' }, i) => (
+        <>
```
Comment (piekczyk) [inline components/DetailsSectionNotification.tsx:57]: key is not defined so I guess there will be warning in console

File: components/DetailsSection.tsx [hunk 2]
```diff
@@ -12,44 +16,66 @@ interface DetailsSectionProps {
...
-export function DetailsSection({ title, badge, buttons, content, footer }: DetailsSectionProps) {
+export function DetailsSection({
+  badge,
+  buttons,
+  content,
+  footer,
+  notifications,
+  title,
+}: DetailsSectionProps) {
+  const [openedNotifications, setOpenedNotifications] = useState<number>(notifications?.length || 0)
```
Comment (piekczyk) [inline components/DetailsSection.tsx:30]: Based on that I assume that every time user switch tab, notifications will pop-up again in details section. Is that expected?

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: components/DetailsSectionNotification.tsx (hunk 1 above)
Comment (piotrkonowrocki) [reply components/DetailsSectionNotification.tsx]: Fixed.

File: components/DetailsSection.tsx (hunk 2 above)
Comment (piotrkonowrocki) [reply components/DetailsSection.tsx:30]: Probably not, closed notifications are stored in `sessionStorage`.

**Review by piekczyk — APPROVED** [review]

**Pull request merged**